consoles/video_stream.pm

```diff
@@ -0,0 +1,353 @@
+    }
+    return $ret;
+}
+
+sub _v4l2_ctl ($device, $cmd) {
+    my @cmd = ("v4l2-ctl", "-d", $device, "-k");
+    push(@cmd, split(/ /, $cmd));
+    my $pipe;
+    my $pid = open($pipe, '-|', @cmd) or return undef;
+    $pipe->read(my $str, 50);
+    my $ret = waitpid($pid, 0);
+    if ($ret > 0 && $? == 0) {
+        # remove header and whitespaces
+        $str =~ s/DV timings://;
+        $str =~ s/^\s+|\s+$//g;
+        return $str;
+    }
+    return undef;
+}
+
+sub connect_remote ($self, $args) {
+    $self->{_last_update_received} = 0;
+
+    if ($args->{url} =~ m/^\/dev\/video/) {
+        if ($args->{edid}) {
+            my $ret = _v4l2_ctl($args->{url}, "--set-edid $args->{edid}");
+            die "Failed to set EDID" unless defined $ret;
+        }
+
+        my $timings = _v4l2_ctl($args->{url}, '--get-dv-timings');
+        if ($timings) {
+            if ($timings ne "0x0pnan") {
+                $self->{dv_timings} = $timings;
+            } else {
+                $self->{dv_timings} = '';
+            }
+            $self->{dv_timings_supported} = 1;
+            $self->{dv_timings_last_check} = time;
+            bmwqemu::diag "Current DV timings: $timings";
+        } else {
+            $self->{dv_timings_supported} = 0;
+            bmwqemu::diag "DV timings not supported";
+        }
+    } else {
+        # applies to v4l only
+        $self->{dv_timings_supported} = 0;
+    }
+
+    bmwqemu::diag "Starting receiving video stream at $args->{url}";
+    $self->connect_remote_video($args->{url});
+
+    $self->connect_remote_input($args->{input_cmd}) if $args->{input_cmd};
+}
+
+sub _get_ffmpeg_cmd ($self, $url) {
+    my @cmd = ('/usr/bin/ffmpeg', '-loglevel', 'fatal', '-i', $url);
+    push(@cmd, ('-vcodec', 'ppm', '-f', 'rawvideo', '-r', '2', '-'));
+    return \@cmd;
+}
+
+sub connect_remote_video ($self, $url) {
+    if ($self->{dv_timings_supported}) {
+        if (!_v4l2_ctl($url, '--set-dv-bt-timings query')) {
+            bmwqemu::diag("No video signal");
+            return;
+        }
+        $self->{dv_timings} = _v4l2_ctl($url, '--get-dv-timings');
+    }
+
+    my $cmd = $self->_get_ffmpeg_cmd($url);
+    my $ffmpeg;
+    $self->{ffmpegpid} = open($ffmpeg, '-|', @$cmd)
+      or die "Failed to start ffmpeg for video stream at $url";
+    $self->{ffmpeg} = $ffmpeg;
+    $ffmpeg->blocking(0);
+
+    $self->{_last_update_received} = time;
+
+    return 1;
+}
+
+sub connect_remote_input ($self, $cmd) {
+    $self->{mouse} = {x => -1, y => -1};
+
+    bmwqemu::diag "Connecting input device";
+
+    my $input_pipe;
+    $self->{inputpid} = open($input_pipe, '|' . $cmd)
+      or die "Failed to start input_cmd($cmd)";
+    $self->{input_pipe} = $input_pipe;
+    $self->{input_pipe}->autoflush(1);
+
+    return $input_pipe;
+}
+
+
+sub _receive_frame ($self) {
+    my $ffmpeg = $self->{ffmpeg};
+    $ffmpeg or die 'ffmpeg does not exist. Probably your backend instance could not start or died.';
```

Comment:
`exist` makes me think not installed, I assume as the message says this is about the process aborting

```suggestion
    $ffmpeg or die 'ffmpeg is not running. Probably your backend instance could not start or died.';
```